src/compiler/checker.ts

```diff
@@ -981,6 +982,29 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsValue(errorLocation: Node, name: string, meaning: SymbolFlags, nameNotFoundMessage: DiagnosticMessage, nameArg: string | Identifier): boolean {
+            const strictlyValueMeanings = SymbolFlags.Value & ~SymbolFlags.Type;
+            const strictlyTypeMeanings = SymbolFlags.Type & ~SymbolFlags.Value;
+
+            if (!(meaning & strictlyValueMeanings) || meaning & SymbolFlags.NamespaceModule) {
+                return false;
+            }
+
+            const nameAsType = resolveName(errorLocation, name, strictlyTypeMeanings, nameNotFoundMessage, nameArg);
+            if (!nameAsType) {
+                return false;
+            }
+
+            if (nameAsType.flags & SymbolFlags.Alias) {

```

Comment:
just call resolveName with`~NamespaceModule`, resolveName does this alias resolution already.